nengo/networks/tests/test_actionselection.py

```diff
@@ -46,3 +45,72 @@ def test_thalamus(Simulator, plt, seed):
 
     assert output[0] > 0.8
     assert np.all(output[1:] < 0.01)
+
+
+def test_bias_input(Simulator, plt, seed):
+    with nengo.Network(seed=seed) as net:
+        bg = nengo.networks.BasalGanglia(dimensions=3, input_bias=-0.5)
+        input = nengo.Node([0.8, 0.5, 0.2], label="input")
+        nengo.Connection(input, bg.input, synapse=None)
+        p_in = nengo.Probe(bg.input, synapse=0.01)
+
+    with Simulator(net) as sim:
+        sim.run(0.1)
+
+    t = sim.trange()
+    plt.plot(t, sim.data[p_in])
+    plt.ylabel("Input")
+
+    assert np.allclose(sim.data[p_in][t > 0.08], [0.3, 0, -0.3], atol=0.005)
+
+
+def test_overridden_configs():
+    # Using default synapses, all that are not None should be
+    # Lowpass with tau of 0.002 (AMPA) or 0.008 (GABA)
+    bg = nengo.networks.BasalGanglia(dimensions=2)
+    for connection in bg.all_connections:
+        if connection.synapse is None:
+            continue
+        assert isinstance(connection.synapse, nengo.Lowpass)
+        assert connection.synapse.tau in (0.002, 0.008)
+
+    # This config does not override the synapse on the connection.
+    # The synapses should be the same as before, and the config should
+    # be unchanged after instantiation, even though inside the class
+    # it is changed.
+    config_no_synapse = nengo.Config(nengo.Connection)
+    config_no_synapse[nengo.Connection].solver = nengo.solvers.LstsqL2nz()
+    assert "synapse" not in config_no_synapse[nengo.Connection]
+
+    bg = nengo.networks.BasalGanglia(
+        dimensions=2,
+        ampa_config=config_no_synapse,
+        gaba_config=config_no_synapse,
+    )
+    for connection in bg.all_connections:
+        if connection.synapse is None:
+            continue
+        # Same synapses
+        assert isinstance(connection.synapse, nengo.Lowpass)
+        assert connection.synapse.tau in (0.002, 0.008)
+        # But solver should be different
+        assert isinstance(connection.solver, nengo.solvers.LstsqL2nz)
+    # Synapse should still not be in the config
+    assert "synapse" not in config_no_synapse[nengo.Connection]
+
+    # This config overrides the synapse on AMPA and GABA connections
+    config_with_synapse = nengo.Config(nengo.Connection)
+    config_with_synapse[nengo.Connection].synapse = nengo.Alpha(0.1)
+    bg = nengo.networks.BasalGanglia(
+        dimensions=2,
+        ampa_config=config_with_synapse,
+        gaba_config=config_with_synapse,
+    )
+    for connection in bg.all_connections:
+        if connection.synapse is None:
+            continue
+        # Synapse changes
+        assert isinstance(connection.synapse, nengo.Alpha)
+        assert connection.synapse.tau == 0.1
+        # Solver should be the same
+        assert not isinstance(connection.solver, nengo.solvers.LstsqL1)
```

Comment:
We should check that the correct synapse is still in the config at the end (it hasn't been overwritten or removed).